src/main/java/io/fusionauth/samlv2/domain/Assertion.java

```diff
@@ -43,10 +44,7 @@ public Assertion(Assertion other) {
         .collect(
             Collectors.toMap(
                 Entry::getKey,
-                entry -> entry.getValue()
-                              .stream()
-                              .map(String::new)
-                              .toList()
+                entry ->  new ArrayList<>(entry.getValue())
```

Comment:
Should a `null` value end up as a `null` in a collection, or just a `null` value? 

And if I use this copy constructor, should we try adding an `equals` in the test to ensure they are still equal?  This would in theory prove the copy constructor didn't change the value - at least as defined by the `equals` and `hashCode` methods.